engine/order_manager_test.go

```diff
@@ -555,6 +597,58 @@ func TestSubmit(t *testing.T) {
 	}
 }
 
+func TestOrderManager_Modify(t *testing.T) {
+	m := OrdersSetup(t)
+	pair := currency.Pair{
+		Base:  currency.NewCode("XXXXX"),
+		Quote: currency.NewCode("YYYYY"),
+	}
+	err := m.orderStore.add(&order.Detail{
+		Exchange:  testExchange,
+		AssetType: asset.Spot,
+		Pair:      pair,
+		ID:        "fake_order_id",
+
+		Price:  10,
+		Amount: 20,
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	mod := order.Modify{
+		// These fields identify the order.
+		Exchange:  testExchange,
+		AssetType: asset.Spot,
+		Pair:      pair,
+		ID:        "nonexistent_order_id",
+
+		// These fields modify the order.
+		Price:  11,
```

Comment:
With the `!= 0` check in mind, can you add a scenario where price is 0 and another where amount is 0?